# 2025.09.10 10.03.13 update batch2 manifest with genotype.rmarkdown
Holly Beale
2025-09-10

``` r
library(tidyverse)
```

    Warning: package 'readr' was built under R version 4.2.3

    Warning: package 'dplyr' was built under R version 4.2.3

    ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
    ✔ dplyr     1.1.4     ✔ readr     2.1.5
    ✔ forcats   1.0.0     ✔ stringr   1.5.0
    ✔ ggplot2   3.4.4     ✔ tibble    3.2.1
    ✔ lubridate 1.9.4     ✔ tidyr     1.3.1
    ✔ purrr     1.0.2     
    ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
    ✖ dplyr::filter() masks stats::filter()
    ✖ dplyr::lag()    masks stats::lag()
    ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors

``` r
library(janitor)
```


    Attaching package: 'janitor'

    The following objects are masked from 'package:stats':

        chisq.test, fisher.test

``` r
library(here)
```

    here() starts at /Users/hbeale/Documents/Dropbox/ucsc/projects/gitCode/splicedice_analysis

``` r
batch_2_bam_manifest_without_gt <- 
  read_tsv(here("2025-05_tcga_luad_download/tcga_bams.50_samples.batch2.2025.05.29_16.01.35.txt"),
           col_names = c("sample_name",
                         "bam_file_path",
                         "id1",
                         "id2")) %>%
  mutate(sample_name = str_replace_all(sample_name, " ", ""))
```

    Rows: 46 Columns: 4
    ── Column specification ────────────────────────────────────────────────────────
    Delimiter: "\t"
    chr (4): sample_name, bam_file_path, id1, id2

    ℹ Use `spec()` to retrieve the full column specification for this data.
    ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
# batches_1_and_2_bed_manifest_without_gt <- read_tsv(here("2025-05_tcga_luad_download/batches_1_and_2_bed_manifest.2025.05.29_22.26.44.txt"),
#                                                     col_names = c("sample_name",
#                                                                   "bed_file_path",
#                                                                   "id1",
#                                                                   "id2")) %>%
#   mutate(sample_name = str_replace_all(sample_name, " ", "_"))

luad_u2af1_s34f_wt_manifest  <- read_tsv(here("2025-05_tcga_luad_sig_from_bam/luad_u2af1_s34f_wt_manifest.tsv"),
                                        col_names = c("id", "mutation"))
```

    Rows: 532 Columns: 2
    ── Column specification ────────────────────────────────────────────────────────
    Delimiter: "\t"
    chr (2): id, mutation

    ℹ Use `spec()` to retrieve the full column specification for this data.
    ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
bam_manifest_with_bam_file_uuid <- batch_2_bam_manifest_without_gt %>%
  mutate(bam_file_uuid = str_remove(bam_file_path, "^.*/") %>% str_remove(".rna_seq.*$")) 

table(bam_manifest_with_bam_file_uuid$bam_file_uuid %in% luad_u2af1_s34f_wt_manifest$id)
```


    TRUE 
      46 

``` r
batch_2_bam_manifest_with_gt <-   left_join(
  bam_manifest_with_bam_file_uuid,
  luad_u2af1_s34f_wt_manifest,
  by=c("bam_file_uuid" = "id"))

# set.seed(133455566)
# batch_2_bam_manifest_with_gt_distinct <- batch_2_bam_manifest_with_gt %>%
#   group_by(bam_file_uuid) %>%
#   slice_sample(n = 1)
# 
# 
# 
#   
# 
# nrow(batch_2_bam_manifest_with_gt) 
# nrow(batch_2_bam_manifest_with_gt_distinct)
```

``` r
head(batch_2_bam_manifest_with_gt)
```

    Warning: 'xfun::attr()' is deprecated.
    Use 'xfun::attr2()' instead.
    See help("Deprecated")

| sample_name | bam_file_path | id1 | id2 | bam_file_uuid | mutation |
|:---|:---|:---|:---|:---|:---|
| TCGA-67-6215-01A0a26152a-462f-4895-8fe8-15fcdcc56e16 | /mnt/data/tcga/0a26152a-462f-4895-8fe8-15fcdcc56e16/7a7440bf-1ca1-4c6b-80f8-7151a38e5d18.rna_seq.genomic.gdc_realn.bam | unknown | solid_tissue | 7a7440bf-1ca1-4c6b-80f8-7151a38e5d18 | u2af1-wt |
| TCGA-86-8075-01A0c633b9e-3303-4625-b59d-02102d8bf981 | /mnt/data/tcga/0c633b9e-3303-4625-b59d-02102d8bf981/5158a031-b856-4423-9418-031b3107e88f.rna_seq.genomic.gdc_realn.bam | unknown | solid_tissue | 5158a031-b856-4423-9418-031b3107e88f | u2af1-wt |
| TCGA-49-4505-01A0ebf5cc5-f242-45ef-821a-939b51dc95a2 | /mnt/data/tcga/0ebf5cc5-f242-45ef-821a-939b51dc95a2/330845b9-1d53-47af-8cb7-30ce5d30625d.rna_seq.genomic.gdc_realn.bam | unknown | unknown | 330845b9-1d53-47af-8cb7-30ce5d30625d | u2af1-s34f |
| TCGA-64-1680-01A16b44441-90d4-4289-8248-d31251f49f2b | /mnt/data/tcga/16b44441-90d4-4289-8248-d31251f49f2b/8b3aec43-4c75-4598-bf0a-168f7ffb9f3b.rna_seq.genomic.gdc_realn.bam | unknown | solid_tissue | 8b3aec43-4c75-4598-bf0a-168f7ffb9f3b | u2af1-s34f |
| TCGA-44-7659-01A20592e25-4b12-4cd3-b1b1-b8e8d6352960 | /mnt/data/tcga/20592e25-4b12-4cd3-b1b1-b8e8d6352960/a0c9ebdf-951b-4d8d-aa79-3ff1c82342cd.rna_seq.genomic.gdc_realn.bam | unknown | solid_tissue | a0c9ebdf-951b-4d8d-aa79-3ff1c82342cd | u2af1-wt |
| TCGA-55-6985-01A2a600a38-215c-4b2b-9b9d-ab5d3b9a0bbc | /mnt/data/tcga/2a600a38-215c-4b2b-9b9d-ab5d3b9a0bbc/dc3ac0d3-d862-4761-ac12-60bb4d6758c1.rna_seq.genomic.gdc_realn.bam | unknown | unknown | dc3ac0d3-d862-4761-ac12-60bb4d6758c1 | u2af1-wt |

check features

``` r
gdc_sample_sheet <- read_tsv(here("2025-05_tcga_luad_sig_from_bam/gdc_sample_sheet.2025-05-22.tsv"))
```

    Rows: 541 Columns: 11
    ── Column specification ────────────────────────────────────────────────────────
    Delimiter: "\t"
    chr (11): File ID, File Name, Data Category, Data Type, Project ID, Case ID,...

    ℹ Use `spec()` to retrieve the full column specification for this data.
    ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
lc_no_space <- function(x) str_replace_all(x, " ", "_") %>% tolower()
gdc_sample_sheet_renamed <- gdc_sample_sheet %>% rename_with(lc_no_space)

head(gdc_sample_sheet_renamed)
```

    Warning: 'xfun::attr()' is deprecated.
    Use 'xfun::attr2()' instead.
    See help("Deprecated")

| file_id | file_name | data_category | data_type | project_id | case_id | sample_id | tissue_type | tumor_descriptor | specimen_type | preservation_method |
|:---|:---|:---|:---|:---|:---|:---|:---|:---|:---|:---|
| 704d42ab-3e11-4c3b-a74c-b1e6a30e27c5 | 325c7d6b-292d-46e5-85d1-785ae8a48c33.rna_seq.genomic.gdc_realn.bam | Sequencing Reads | Aligned Reads | TCGA-LUAD | TCGA-44-2659 | TCGA-44-2659-01A | Tumor | Primary | Solid Tissue | Unknown |
| d1945e55-eaa9-41f3-8017-380ccd112dfc | 59a0b90e-8b8b-41b8-95f0-d51961a94be5.rna_seq.genomic.gdc_realn.bam | Sequencing Reads | Aligned Reads | TCGA-LUAD | TCGA-05-4396 | TCGA-05-4396-01A | Tumor | Primary | Solid Tissue | Unknown |
| 44a3eb8c-135f-44f4-82bd-86fb6104a4e8 | f3d482bb-14e8-4569-bee4-22d7d2a027ea.rna_seq.genomic.gdc_realn.bam | Sequencing Reads | Aligned Reads | TCGA-LUAD | TCGA-55-7725 | TCGA-55-7725-01A | Tumor | Primary | Solid Tissue | Unknown |
| 216b822e-0d33-476f-ba03-18ff818f8a78 | 5fd9d2a8-3a95-440c-a0de-8ad5fd86e156.rna_seq.genomic.gdc_realn.bam | Sequencing Reads | Aligned Reads | TCGA-LUAD | TCGA-05-5420 | TCGA-05-5420-01A | Tumor | Primary | Solid Tissue | Unknown |
| ba6057cc-61fd-4d2f-8599-ed0a6aaf80b6 | 145db6c5-f7ad-4e2f-88f1-d68bc457848a.rna_seq.genomic.gdc_realn.bam | Sequencing Reads | Aligned Reads | TCGA-LUAD | TCGA-97-8172 | TCGA-97-8172-01A | Tumor | Primary | Solid Tissue | Unknown |
| 45600709-c917-42ef-bc2f-229e6f3c71af | 9ebc079c-069f-4468-8b9e-680e39a3c4ad.rna_seq.genomic.gdc_realn.bam | Sequencing Reads | Aligned Reads | TCGA-LUAD | TCGA-38-4629 | TCGA-38-4629-01A | Tumor | Primary | Unknown | Unknown |

``` r
batch_2_bam_manifest_with_gt_anno <- left_join(batch_2_bam_manifest_with_gt %>%
                                                                  mutate(gdc_file_uuid = str_remove(sample_name, "^.*_")),
          gdc_sample_sheet_renamed, by=c("gdc_file_uuid" = "file_id"))
```

# review sample features

``` r
n_unique <- function(x) length(unique(x))

unique_sample_vals <- batch_2_bam_manifest_with_gt_anno %>%
#   mutate(across(everything(), as.character)) %>%
  ungroup %>%
  summarize(across(everything(),  n_unique)) %>%
  pivot_longer(everything())

unique_sample_vals
```

    Warning: 'xfun::attr()' is deprecated.
    Use 'xfun::attr2()' instead.
    See help("Deprecated")

| name                | value |
|:--------------------|------:|
| sample_name         |    46 |
| bam_file_path       |    46 |
| id1                 |     2 |
| id2                 |     2 |
| bam_file_uuid       |    46 |
| mutation            |     2 |
| gdc_file_uuid       |    46 |
| file_name           |     1 |
| data_category       |     1 |
| data_type           |     1 |
| project_id          |     1 |
| case_id             |     1 |
| sample_id           |     1 |
| tissue_type         |     1 |
| tumor_descriptor    |     1 |
| specimen_type       |     1 |
| preservation_method |     1 |

``` r
tabyl(batch_2_bam_manifest_with_gt_anno,
      specimen_type)
```

    Warning: 'xfun::attr()' is deprecated.
    Use 'xfun::attr2()' instead.
    See help("Deprecated")

| specimen_type |   n | percent | valid_percent |
|:--------------|----:|--------:|--------------:|
| NA            |  46 |       1 |            NA |

``` r
tabyl(batch_2_bam_manifest_with_gt_anno,
      preservation_method)
```

    Warning: 'xfun::attr()' is deprecated.
    Use 'xfun::attr2()' instead.
    See help("Deprecated")

| preservation_method |   n | percent | valid_percent |
|:--------------------|----:|--------:|--------------:|
| NA                  |  46 |       1 |            NA |

consider removing OCT

``` r
batch_2_bam_manifest_with_gt_anno %>%
  filter(preservation_method == "OCT")
```

    Warning: 'xfun::attr()' is deprecated.
    Use 'xfun::attr2()' instead.
    See help("Deprecated")

| sample_name | bam_file_path | id1 | id2 | bam_file_uuid | mutation | gdc_file_uuid | file_name | data_category | data_type | project_id | case_id | sample_id | tissue_type | tumor_descriptor | specimen_type | preservation_method |
|:---|:---|:---|:---|:---|:---|:---|:---|:---|:---|:---|:---|:---|:---|:---|:---|:---|

no, one is mutant, so i’ll keep all 4

# generate bam manifest output

``` r
batch_2_bam_manifest_with_gt_anno %>%
  ungroup %>%
  mutate(id1 = mutation,
         id2 = mutation) %>% 
  select(1:4) %>%
write_tsv(here("2025-05_tcga_luad_download/batch_2_bam_manifest.with_genotypes.2025.09.10_10.03.13.tsv"),
            col_names = FALSE)
```
